File: src/088-Merge-Sorted-Array/088_merge_sorted_array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int index_nums1 = m - 1;
        int index_nums2 = n - 1;
        int icur = m + n - 1;

        while (index_nums1 >= 0 && index_nums2 >= 0) {
            nums1[icur--] = (nums1[index_nums1] >= nums2[index_nums2] ? nums1[index_nums1--] : nums2[index_nums2--]);
        }
        while (index_nums2 >= 0) {
            nums1[icur--] = nums2[index_nums2--];
        }
    }
};

File: src/088-Merge-Sorted-Array/088_merge_sorted_array.py
class Solution:
    def merge(self, nums1, m, nums2, n):
        """
        Given two sorted integer arrays nums1 and nums2, merge nums2 into nums1 as one sorted array.

        Args:
            nums1(List[int]): the name of the integer array nums1
            m(int): the size(length) of the integer array nums1 
            nums2(List[int]): the name of the integer array nums2
            n(int): the size(length) of the integer array nums2

        Returns:
            void Do not return anything, modify nums1 in-place instead.
        """
        while m > 0 and n > 0:
            if nums1[m-1] >= nums2[n-1]:
                nums1[m+n-1] = nums1[m-1]
                m -= 1
            else:
                nums1[m+n-1] = nums2[n-1]
                n -= 1
        if n > 0:
            nums1[:n] = nums2[:n]
